src/dune_rules/lib.ml

```diff
@@ -1081,6 +1093,16 @@ end = struct
   let resolve_simple_deps db names ~private_deps =
     Resolve.Memo.List.map names ~f:(resolve_dep db ~private_deps)
 
+  let resolve_ppx_runtime_deps db names ~private_deps ~for_lib =
```

Comment:
Instead of passing `for_lib` everywhere, how about just adding the stack frame before the call that resolves the runtime deps?